app/jobs/cleanup_account_job.rb

```diff
@@ -1,18 +1,27 @@
 class CleanupAccountJob < ActiveJob::Base
   def perform(account)
-    cleanup_solr(account)
-    cleanup_fedora(account)
+    account.switch do
+      cleanup_solr(account)
+      cleanup_fedora(account)
+      # Redis doesn't need account, as account.switch changes Redis to correct namespace
+      cleanup_redis
+    end
     Apartment::Tenant.drop(account.tenant)
     account.destroy
   end
 
   private
 
     def cleanup_fedora(account)
-      account.switch do
-        # Preceding slash must be removed from base_path when calling delete()
-        fcrepo_client.delete(account.fcrepo_endpoint.base_path.sub!(%r{^\/}, ''))
-      end
+      # Preceding slash must be removed from base_path when calling delete()
+      fcrepo_client.delete(account.fcrepo_endpoint.base_path.sub!(%r{^\/}, ''))
+    end
+
+    def cleanup_redis
+      # Redis::Namespace currently doesn't support flushall or flushdb.
+      # So, we must loop through and delete all keys in this namespace.
+      # See https://github.com/resque/redis-namespace/issues/56
+      redis_namespace.keys.map { |key| redis_namespace.del(key) }
```

Comment:
@mjgiarlo That is an inappropriate use of map.  Map is supposed to produce a transformation for each input, not be a disguised form of loop.